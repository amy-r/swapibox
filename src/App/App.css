.App {
  display: grid;
  grid-template-columns: repeat(5,1fr);
  grid-template-rows: repeat(3,1fr);
  grid-gap: 10px;
}

.Scroll {
  flex-basis: 100%;
}

header {
  grid-column: 1/6;
  display: inline-flex; 
  justify-content: space-between;
  background-color: #000;
}

h2 {
  color: white;
  font-size: 36px;
  letter-spacing: 3px;
  text-transform: uppercase;
}

.button-section {
  grid-column: 1/6;
  grid-row: 2;
  display: flex;
  justify-content: space-around;
}
