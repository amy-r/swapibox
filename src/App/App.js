import React, { Component } from 'react';
import './App.css';
import Card from '../Card/Card.js';
import CardContainer from '../CardContainer/CardContainer.js'
import FavoritesButton from '../FavoritesButton/FavoritesButton.js';
import PeopleButton from '../PeopleButton/PeopleButton.js';
import PlanetsButton from '../PlanetsButton/PlanetsButton.js';
import ScrollText from '../ScrollText/ScrollText.js';
import VehiclesButton from '../VehiclesButton/VehiclesButton.js';
import './App.css';
import {getPeople, getPlanets, getVehicles} from '../helper.js'

class App extends Component {
  constructor(){

    super();
    this.state = {
      people: [],
      planets: [],
      vehicles: [],
      favorites: [],
      current: 'people',
      errorStatus: ''
    }
  }


clickFunction = async (e) => {
  console.log(e.target.className)
  if (e.target.className === 'PeopleButton'){
    const people = await getPeople();
    
    this.setState({
      people: people,
      current: 'people'
    })
  }
  else if (e.target.className ==='PlanetsButton'){
    const planets = await getPlanets();

    this.setState({
      planets: planets,
      current: 'planets'
    })
  }
  else if (e.target.className ==="VehiclesButton"){
    const vehicles = await getVehicles();

    this.setState({
      vehicles: vehicles,
      current: 'vehicles'
    })
  }
}
  
async componentDidMount() {
    // await Helper.getPeople();
    // await Helper.getPlanets();
    // await Helper.getVehicles();
  }

  render() {
    return (
      <div className="App">
        <header>
          <h2> SWAPI-Box </h2> 
          <FavoritesButton /> 
        </header>  
        <div className= 'button-section'>
          <PeopleButton onClick= {this.clickFunction}/> <PlanetsButton onClick= {this.clickFunction}/> <VehiclesButton onClick= {this.clickFunction}/> 
        </div>
        <CardContainer display={this.state[this.state.current]}/>
        <ScrollText className="Scroll"/>
      </div>
    );
  }

}
export default App;
