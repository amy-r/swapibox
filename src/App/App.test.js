
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe( 'App Component', () => {

  it('exists', () => {
  })

  it('renders without crashing', () => {
    const renderedComponent = shallow(<App />, {disableLifecycleMethods: true})
    expect(renderedComponent).toMatchSnapshot();
  })

});