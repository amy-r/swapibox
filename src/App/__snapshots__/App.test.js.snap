// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h2>
                     SWAPI-Box 
          </h2>
          <FavoritesButton />
</header>,
        <div
          className="button-section"
>
          <PeopleButton
                    onClick={[Function]}
          />
           
          <PlanetsButton
                    onClick={[Function]}
          />
           
          <VehiclesButton
                    onClick={[Function]}
          />
</div>,
        <CardContainer
          display={Array []}
/>,
        <ScrollText
          className="Scroll"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
               SWAPI-Box 
</h2>,
            <FavoritesButton />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " SWAPI-Box ",
            },
            "ref": null,
            "rendered": " SWAPI-Box ",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PeopleButton
              onClick={[Function]}
/>,
            " ",
            <PlanetsButton
              onClick={[Function]}
/>,
            " ",
            <VehiclesButton
              onClick={[Function]}
/>,
          ],
          "className": "button-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "display": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "Scroll",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h2>
                         SWAPI-Box 
            </h2>
            <FavoritesButton />
</header>,
          <div
            className="button-section"
>
            <PeopleButton
                        onClick={[Function]}
            />
             
            <PlanetsButton
                        onClick={[Function]}
            />
             
            <VehiclesButton
                        onClick={[Function]}
            />
</div>,
          <CardContainer
            display={Array []}
/>,
          <ScrollText
            className="Scroll"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                 SWAPI-Box 
</h2>,
              <FavoritesButton />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " SWAPI-Box ",
              },
              "ref": null,
              "rendered": " SWAPI-Box ",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PeopleButton
                onClick={[Function]}
/>,
              " ",
              <PlanetsButton
                onClick={[Function]}
/>,
              " ",
              <VehiclesButton
                onClick={[Function]}
/>,
            ],
            "className": "button-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "display": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "Scroll",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
